crypto/fipsmodule/ec/ec_key.c

```diff
@@ -504,10 +504,18 @@ int EC_KEY_generate_key(EC_KEY *key) {
 }
 
 int EC_KEY_generate_key_fips(EC_KEY *eckey) {
+  int ret = 0;
+  int num_attempts = 0;
   // We have to verify both |EC_KEY_generate_key| and |EC_KEY_check_fips| both
   // succeed before updating the indicator state, so we lock the state here.
   FIPS_service_indicator_lock_state();
-  if (EC_KEY_generate_key(eckey) && EC_KEY_check_fips(eckey)) {
+  do {
+    ret = EC_KEY_generate_key(eckey);
+    ret &= EC_KEY_check_fips(eckey);
+    num_attempts++;
+  } while (ret == 0 && num_attempts < MAX_PCT_ATTEMPTS);
+
+  if (ret) {
```

Comment:
nit: state should be unlocked above this if. We can try to to remember when reviewing server indicator again in the future.